### ISSUE TITLE
dir/server.Lookup: no root for user

### ISSUE THREAD

**Issue by BHare1985** [upspin/upspin#665]
Wow upspin setup and configuration is very confusing and non-forgiving. The wiki isn't really up to date with things like suggesting `go build` not working as expected. Anyways, I got it all running with using WSL ubuntu after a bunch of trial and error and then I was testing the `packing` config  switching from `ee` to `eeintegrity` -- eventually it must of corrupted my server by switching packing with files still present.

No problem, I'll just delete everything and start from a brand new WSL instance because I wanted to document the exact set up I did. I figured since I still had the TXT record on my domain that not having the private keys I'd be OK. Perhaps that was the case before, but I believe since https://github.com/upspin/upspin/issues/249 it is not allowed anymore.  I also thought maybe because I still had access to the email address I used that it could be restored, but per https://github.com/upspin/upspin/issues/614 that is not allowed.

Alright well I guess that user is a lost cause, so I generated a new email (email2@fake.com). Go through all the steps and turn on the server once again, then I go to `upspin ls` and I get

```
upspin: ls: dir/remote("upspin.fake.com:443").Lookup: item does not exist:
        dir/server.Lookup:
        user email2@fake.com:
        no root for user
```

Not sure what the issue is, here are my complete steps (they may differ from the wiki, since its not meant for WSL and/or seems outdated)

```
[Client]

sudo apt-get install golang host
cd ~
go install upspin.io/cmd/{upspin,cacheserver,upspin-audit,upspinfs}@latest
mv ~/go/bin/* /usr/bin/
upspin signup -server=upspin.fake.com email2@fake.com
upspin setupdomain -domain=fake.com
upspin setupserver -domain=fake.com -host=upspin.fake.com




[Server]

sudo apt-get install golang libcap2-bin
sudo useradd -m upspin
sudo usermod --shell /bin/bash upspin 
su upspin
cd ~

mkdir -p upspin/letsencrypt
go install upspin.io/cmd/upspinserver@latest
mv ~/go/bin/upspinserver ~/upspinserver
sudo setcap cap_net_bind_service=+ep ~/upspinserver
~/upspinserver
```

it's worth noting after the `upspin setupserver` command I get
```
User "upspin@fake.com" already exists on key server.
upspin: setupserver: upspinserver returned status 404 Not Found:
404 page not found
```
Which I believe is expected?  When I visit the page `upspin.fake.com` it gives me a 404

**Comment by BHare1985** [comment]
I have high confidence that the 404 not found from `setupserver` is related and causing the issue based on the code I read. The server does adds writers with the log saying `Setting writers to: [email2@fake.com upspin@fake.com]` but seems never creates a root, which follows the code logic. This is where the 404 is throwing I believe; https://github.com/upspin/upspin/blob/70e5bc8005f9c37cc362db0ef214b3a83c1c4b8f/cmd/upspin/setupserver.go#L181  because when I visit `https://upspin.fake.com/setupserver` is gives a 404.

**Comment by BHare1985** [comment]
Welp I solve the issue by deleting the `upspin/server` directory on the server, then re-running `setupserver` on the client as per https://upspin.io/doc/server_setup.md 

Curiously, the `no root user` error appeared on the server output:

upspin setupserver output:
```
User "upspin@fake.com" already exists on key server.
Configured upspinserver at "upspin.fake.com:443".
Created root "email2@fake.com".
```
upspinserver output:
```
2024/07/25 15:47:06.067097 Store and Directory servers initialized.
2024/07/25 15:47:06.067168 Store server configuration: backend=Disk basePath=/home/upspin/upspin/server/storage
2024/07/25 15:47:06.148213 serverutil/perm.NewFromDir: item does not exist:
        dir/server.Watch:
        user upspin@fake.com:
        no root for user
```

but as far as I can tell it's working properly, as I can put/get without issue.

**Comment by BHare1985** [comment]
TLDR:

Issue was that the server must of been setup with a different user, and once you setup once you cannot setupserver again (you get a 404 if you try a second time). If you lose the client data, you aren't able to create another user to replace it (either via setupserver or setupwriters).  You should delete the `server` folder on the server and re-setup. I believe this scenario could use better error handling and message output to make this more clear.

---------

More investigating seems to suggest that once the server is setup, the `/setupserver` endpoint is removed. I believe the client should detect if setupserver has already been ran, and then indicate such in the console instead of showing a 404.

I found this out because I deleted the config setup for `email2@fake.com`  and created a new user `email3@fake.com` and ran through the following commands:

```
upspin signup -server=upspin.fake.com email3@fake.com
upspin setupdomain -domain=fake.com
upspin setupserver -domain=fake.com -host=upspin.fake.com
```

To which it gave the same error again (404 and then no root for email3@fake.com). It seems like once a server is setup you don't go through the same process for a new user if you misplace the initial user.  The reason I attempted it this way is because  you can't use `upspin setupwriters -domain=fake.com email3@fake.com` because it doesn't allow you unless you are already a writer I suppose:

```
upspin: setupwriters: client.MakeDirectory: upspin@fake.com/: I/O error:
        dir/remote("upspin.fake.com:443").Put:
        rpc.Invoke: 401 Unauthorized: rpc.SessionForRequest: permission denied:
        user upspin@fake.com: invalid signature: invalid operation: signature does not match
```
